### PR TITLE
Porting rqt_top to ROS2

### PR DESCRIPTION
**Pull request by brawner**
This ports rqt_top to ROS2. Unfortunately much of the functionality this plugin relies on has yet to be exposed in the client libraries of ROS2 and so it has been commented out. Currently this plugin will only show you a list of running nodes in the graph.

Related issues have been created on ros2/ros2
https://github.com/ros2/ros2/issues/600
https://github.com/ros2/ros2/issues/601

PR for python3-psutil has been created here
https://github.com/ros/rosdistro/pull/19464

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Does it make sense to port this plugin if almost all of its functionality is unavailable?

**Comment by brawner** [comment]
Removing the functionality was pretty trivial, so I thought I would create this port and related issues to track. 

It certainly is not the best demonstration of capabilities of ros2.

**Comment by mlautman** [comment]
Porting is better than not porting. If someone wants to extend it later it is better if they don't have to start from scratch

**Review by mlautman — COMMENTED** [review]
PluginManager._load_plugin() could not load plugin "rqt_top/TOP"

**Comment by brawner** [comment]
@Mike, I just realized this depended on the subclassed plugin_context PR. I bet that will fix your issue.

https://github.com/ros-visualization/rqt/pull/161

File: src/rqt_top/top_plugin.py [hunk 1]
```diff
@@ -51,22 +55,24 @@ def __lt__(self, other):
...
+    #    0: ('cmdline', lambda x: '\n'.join(textwrap.wrap(' '.join(x)))),
+    #    3: ('memory_info', lambda x: ('Resident: %0.2f MiB, Virtual: %0.2f MiB' % (x[0] / 2**20, x[1] / 2**20)))
+    # }
 
     name_filter = re.compile('')
 
     def __init__(self, context):
         super(Top, self).__init__(context)
+
+        self._node = context._node
```
Comment (mlautman) [inline src/rqt_top/top_plugin.py]: same note about context vs node.

File: src/rqt_top/node_info.py [hunk 2]
```diff
@@ -40,57 +33,78 @@
 class NodeInfo(object):
     nodes = dict()
 
+    def __init__(self, node):
+        self._node = node
```
Comment (mlautman) [inline src/rqt_top/node_info.py:37]: Do we want to bypass the context wrapper? (https://github.com/ros-visualization/rqt/pull/161/) Couldn't we just pass in a copy of the context and then use the helper functions? Or maybe make `context._node` a public member variable (`context.node`) and remove the helper functions?

File: src/rqt_top/top_plugin.py [hunk 3]
```diff
@@ -51,22 +55,24 @@ def __lt__(self, other):
 
 
 class Top(Plugin):
-    NODE_FIELDS   = [             'pid', 'get_cpu_percent', 'get_memory_percent', 'get_num_threads']
-    OUT_FIELDS    = ['node_name', 'pid', 'cpu_percent',     'memory_percent',     'num_threads'    ]
+    NODE_FIELDS   = []#           'pid', 'get_cpu_percent', 'get_memory_percent', 'get_num_threads']
+    OUT_FIELDS    = ['node_name',]  # 'pid', 'cpu_percent',     'memory_percent',     'num_threads'    ]
     FORMAT_STRS   = ['%s',        '%s',  '%0.2f',           '%0.2f',              '%s'             ]
     NODE_LABELS   = ['Node',      'PID', 'CPU %',           'Mem %',              'Num Threads'    ]
```
Comment (mlautman) [inline src/rqt_top/top_plugin.py]: trim these as well?

File: src/rqt_top/node_info.py (hunk 2 above)
Comment (brawner) [reply src/rqt_top/node_info.py:37]: I think this comment should go on the referenced PR

File: src/rqt_top/top_plugin.py (hunk 1 above)
Comment (brawner) [reply src/rqt_top/top_plugin.py]: Reverting back to saving the node object.

File: src/rqt_top/top_plugin.py (hunk 3 above)
Comment (brawner) [reply src/rqt_top/top_plugin.py]: Done

**Pull request merged**